src/HealthChecks.OpenIdConnectServer/DiscoveryEndpointResponse.cs

```diff
@@ -0,0 +1,24 @@
+using System.Text.Json.Serialization;
+
+namespace HealthChecks.IdSvr;
+
+internal class DiscoveryEndpointResponse
+{
+    [JsonPropertyName(OidcConstants.ISSUER)]
+    public string Issuer { get; set; } = null!;
+
+    [JsonPropertyName(OidcConstants.AUTHORIZATION_ENDPOINT)]
+    public string AuthorizationEndpoint { get; set; } = null!;
+
+    [JsonPropertyName(OidcConstants.JWKS_URI)]
+    public string JwksUri { get; set; } = null!;
+
+    [JsonPropertyName(OidcConstants.RESPONSE_TYPES_SUPPORTED)]
+    public IEnumerable<string> ResponseTypesSupported { get; set; } = null!;
+
+    [JsonPropertyName(OidcConstants.SUBJECT_TYPES_SUPPORTED)]
+    public IEnumerable<string> SubjectTypesSupported { get; set; } = null!;
+
+    [JsonPropertyName(OidcConstants.ALGORITHMS_SUPPORTED)]
+    public IEnumerable<string> IdTokenSigningAlgValuesSupported { get; set; } = null!;
```

Comment:
```suggestion
    public string[] ResponseTypesSupported { get; set; } = null!;

    [JsonPropertyName(OidcConstants.SUBJECT_TYPES_SUPPORTED)]
    public string[] SubjectTypesSupported { get; set; } = null!;

    [JsonPropertyName(OidcConstants.ALGORITHMS_SUPPORTED)]
    public string[] IdTokenSigningAlgValuesSupported { get; set; } = null!;
```

Reply:
Could you share more thoughts why array is better than IEnumerable here?

Reply:
1. Nothing to abstract here. Internal code.
2. Abstraction gives nothing except additional iterator allocation when using `Contains` etc.

Reply:
Understood, thanks!